.github/workflows/myci.yml

```
@@ -0,0 +1,24 @@
+name: Build To Do app
+on:
+  push:
+    paths-ignore:
+    - '**/README.md'
+  pull_request:
+    paths-ignore:
+    - '**/README.md'
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    env:
+        TOKEN: ${{ secrets.TOKEN }}
+        KEY: ${{ secrets.KEY }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: echo "Building image...."
+    - run: docker build --target test --tag my-test-image .    
```

I think having a separate step to echo what you're doing leads to more confusion rather than less.

Here's a nicer way of clarifying your steps, especially when you are just describing a single line:
```suggestion
    - name: Build image
      run: docker build --target test --tag my-test-image .     
```
This should lead to a nicer breakdown than what you currently have
https://github.com/rajeshdewan/-DevOps-Course-Starter/runs/4105762624?check_suite_focus=true

For the commands below, you could have a single step with name "run tests" where the run command combines multiple shell commands together with `&&`

---

Review comments incorporated